internal/ringbuffer.go

```diff
@@ -139,3 +135,35 @@ func (rp *RingbufferProxy) checkCounts(minCount int32, maxCount int32) (err erro
 	}
 	return
 }
+
+type LazyReadResultSet struct {
+	readCount            int32
+	dataItems            []*Data
+	itemSequences        []int64
+	serializationService *SerializationService
+}
+
+const SEQUENCE_UNAVAILABLE int64 = -1
+
+func NewLazyReadResultSet(readCount int32, items []*Data, itemSeqs []int64, ss *SerializationService) *LazyReadResultSet {
+	return &LazyReadResultSet{readCount: readCount, dataItems: items, itemSequences: itemSeqs, serializationService: ss}
+}
+
+func (rs *LazyReadResultSet) ReadCount() int32 {
+	return rs.readCount
+}
+
+func (rs *LazyReadResultSet) Get(index int32) (result interface{}, err error) {
+	return rs.serializationService.ToObject(rs.dataItems[index])
+}
+
+func (rs *LazyReadResultSet) Sequence(index int32) int64 {
```

Comment:
In java this method throws `UnsupportedOperationException` when `itemSequences`is nil. Would it be better to return an error too in this method ?

Reply:
Yes but in java you don't return a value when you rise an exception but in go you add an error parameter and return the value anyway so there is no point for an extra error parameter